src/group_theory/perm/fin.lean

```diff
@@ -62,3 +63,170 @@ lemma finset.univ_perm_fin_succ {n : ℕ} :
   @finset.univ (perm $ fin n.succ) _ = (finset.univ : finset $ fin n.succ × perm (fin n)).map
   equiv.perm.decompose_fin.symm.to_embedding :=
 (finset.univ_map_equiv_to_embedding _).symm
+
+section cycle
+
+/-! ### `cycle` section
+
+Define the permutations `fin.cycle_all n`, the cycle `(0 1 2 ... (n - 1 : fin n))`
+and `fin.cycle_range i`, the cycle `(0 1 2 ... i)`.
+-/
+
+namespace fin
+
+open equiv.perm
+
+/-- `fin.cycle_all` is the cycle `(0 1 2 ... fin.last n)`.
+
+If `n` is `0` or `1`, this will be the identity permutation,
+otherwise it sends `i` to `i + 1` and `fin.last (n - 1)` to `0`.
+-/
+def cycle_all : Π (n : ℕ), perm (fin n)
```

Comment:
Isn't this 
```
/-- Rotate `fin n` one step to the right. -/
def fin_rotate : Π n, fin n ≃ fin n
| 0 := equiv.refl _
| (n+1) := fin_add_flip.trans (fin_congr (add_comm _ _))
```
defined in `data.equiv.fin`?

Reply:
Nice find. It would be good to prove:
```
section
local attribute [instance] fin.comm_ring
lemma fin_rotate_succ_eq_add_right {n : nat} :
  fin.rotate n.succ = equiv.add_right 1 := sorry
end
```

Reply:
I'm currently testing defining `cycle_range` just by composing `fin.rotate` with `sum_fin_sum_equiv`.

Reply:
Indeed, it is the same. It had so little API that the only reduction in lines was not having to import `data.zmod.basic`.